### ISSUE TITLE
A 'list badges' command would be nice to have

### ISSUE THREAD

**Issue by therealklanni** [hubot-scripts/hubot-ingress#14]
When someone says 'list badges' reply with a list of the badges types (no need to show each rank for ranked badges).

**Comment by AgentNormandy** [comment]
I've written this and tested it on my local ingress community's slack team page. It lists the badge type and then shows the emojii for each level. Pretty new to coffeescript and github so there may have been a better way to code this but I'm attaching the result. Let me know what you think. 

![capture](https://cloud.githubusercontent.com/assets/10510548/5717236/b6e7fb68-9ac2-11e4-89a5-c52b38b81093.PNG)

**Comment by therealklanni** [comment]
Well done. I just have one concern that it's too "spammy", what do you think?

I think my initial thought was just to do `msg.reply badgeList.join ""` (but showing only the level1 badges.

I'm not asking you to go change it now, just want to get your feedback. Too spammy? A bit spammy, but acceptable (worth it for the valuable info)?

**Comment by therealklanni** [comment]
Also, can you submit what you have as a PR so I can take a look? Thanks :+1:

**Comment by drstevens** [comment]
We discovered that only emojis on the last line of messages appear in the Slack android client. Not sure if this matters here and I assume it will eventually be fixed. I have no idea if there is a support ticket open or if they even know about the issue yet.

e.g. in the response above, only the `verified` emoji will show up in the android client

**Comment by therealklanni** [comment]
True, there are a lot of issues with emoji in the Android client. I'm part of their beta, so I will check it on mine and see if there's been any improvement. Feel free to use the `/feedback` command in Slack to ask them about it (perhaps if they're seeing a lot of different people on different teams asking about it, they will bump up the priority?).

Getting back on topic, however, @drstevens what are your thoughts on the above example output for this command? Do you find it spammy or is it acceptable?

**Comment by drstevens** [comment]
I do find it a bit spammy. I think comma separated list of badge names (no emojis) would be more than enough. We have a dedicated channel for people to execute commands spammy commands like this , `help`, and `what do you know` though.

**Comment by AgentNormandy** [comment]
I think it gives a good amount of valuable info by allowing you to see every badge possible in both name and in image, but I agree there is a lot of it. 

Are you thinking just show the badge types and no emoji's just as in a list like:
builder, connector, darsana, engineer, eve,

or as a more detailed list 
builder1, connector1, darsana, engineer1, eve.

**Comment by drstevens** [comment]
I would think just `builder, connector, darsana, ...`

**Comment by therealklanni** [comment]
Well, I hadn't put all that much thought into it. I created the issue just to capture the idea when I came up in a discussion in Slack. I planned on coming back to it at some point, but @AgentNormandy was nice enough to tackle it already (thank you).

But to answer your question, I guess I had envisioned it just being a list of the possible badges (and probably having put more thought into it I would consider some sort of notation to indicate the badges that have tiers).

**Comment by drstevens** [comment]
Personally, I don't really care at all.

**Comment by AgentNormandy** [comment]
I'll throw some stuff at the wall after work and see what sticks.

**Comment by therealklanni** [comment]
@drstevens thanks for your input at least

@AgentNormandy sounds good. And since you already have it running in your Slack, please let us know how well it works with your users. If nobody seems to think it's spammy, then maybe it's fine the way it is. Also, if you send that PR, I can do some testing in mine as well.

**Comment by AgentNormandy** [comment]
Here's what I got, I went with the output formatted as code but not married to that idea. 
![image](https://cloud.githubusercontent.com/assets/10510548/5729038/baa002ce-9b34-11e4-8b57-ab9ed2178cc8.png)

I do miss the emjoi's but I'm a visual guy. What do you think about having the attached output as list badges and having the output I originally posted as display badges.

**Comment by AgentNormandy** [comment]
Hope I did that right, I'm sure there's probably a better way to code what I did so please let me know any suggestions or changes you recommend. Output for current version. 
![image](https://cloud.githubusercontent.com/assets/10510548/5731257/34db6d82-9b47-11e4-9d46-1a5d4de712de.png)

**Comment by AgentNormandy** [comment]
@therealklanni Wanting to make sure I remove that commit correctly, could you shoot me a message on gmail/hangouts/slack? agentnormandy @ gmail

**Comment by therealklanni** [comment]
No problem, just do the following:

```bash
# this will create a new branch starting before your badge-list commits
git checkout -b intel-map 3baf53e^
# this will apply the intel-map commit to the new branch
git cherry-pick 3eed6ff
# now switch back to master
git checkout master
# and remove the intel-map commit
git reset --hard HEAD^
# now you have to force push the change, since you already pushed that commit
# force pushes should be avoided, generally, but in this case necessary
git push origin master -f
```

In the future, always branch before making changes, so you don't get yourself in the same situation again. This helps you keep your changes separate. :metal: 

When you're ready for the intel map changes to be merged, you can push that new branch and do a PR from that.

**Comment by AgentNormandy** [comment]
Done, let me know if I need to do anything else.

**Comment by therealklanni** [comment]
Resolved by #20
